firebase-firestore/src/main/java/com/google/firebase/firestore/core/Query.java

```diff
@@ -325,11 +325,16 @@ public List<OrderBy> getExplicitOrderBy() {
   }
 
   /**
-   * Returns the full list of ordering constraints on the query.
+   * Returns the full list of ordering constraints on the query. This might include additional
+   * sort orders added implicitly to match the backend behavior.
    *
-   * <p>This might include additional sort orders added implicitly to match the backend behavior.
+   * This method is marked as synchronized to prevent the memoizedOrderBy from being
```

Comment:
I think in comments we should avoid talking about implementation details too much, some remarks like "This is synchronized because it actually modifies internal state in some cases" should be good enough.

Reply:
Updated